openedx_events/learning/data.py

```diff
@@ -190,3 +190,29 @@ class CourseDiscussionConfigurationData:
     unit_level_visibility = attr.ib(type=bool, default=False)
     plugin_configuration = attr.ib(type=dict, default={})
     contexts = attr.ib(type=List[DiscussionTopicContext], factory=list)
+
+
+@attr.s(frozen=True)
+class GradeData:
+    """
+    Attributes defined for Open edX grades data object.
+
+    Arguments:
+        user (int): identifier of the grade to which the grade belongs.
+        course_key (CourseData): Identifier of the course to which the grade belongs.
+        course_edited_timestamp (datetime): date the grade was edited.
+        course_version (str): version of the course.
+        grading_policy_hash (str): grading policy hash of the course.
+        percent_grade (float): percentage of the note.
+        letter_grade (str): grade in letter
+        passed_timestamp (datetime): date the course was passed.
+    """
+
+    user_id = attr.ib(type=int)
```

Comment:
I think this raises an interesting question for future devs: should we enforce using existent data classes when possible? should we make all user attributes optional but the ID/username? (because if you're going to have a reference for the user, it's going to be its ID or username)
@felipemontoya

Reply:
I'd make sure that these nuances are documented in the how-tos.